src/model/index.ts

```diff
@@ -215,13 +253,44 @@ class Model implements WorkbookModel {
       ids.add(value.security_id);
       this.issuedSecuritiesByStakeholderAndStockClassIds_.set(key, ids);
     }
+    const txns =
+      this.transactionsBySecurityId_.get(value.security_id) || new Set();
+    txns.add(value);
+    this.transactionsBySecurityId_.set(value.security_id, txns);
+  }
 
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private TX_PLAN_SECURITY(value: any) {
+    if (value.object_type === "TX_PLAN_SECURITY_ISSUANCE") {
+      const key = `${value.stakeholder_id}/${value.stock_plan_id}`;
+      const ids =
+        this.issuedSecuritiesByStakeholderAndStockPlanIds_.get(key) ||
+        new Set();
+      ids.add(value.security_id);
+      this.issuedSecuritiesByStakeholderAndStockPlanIds_.set(key, ids);
+    }
     const txns =
       this.transactionsBySecurityId_.get(value.security_id) || new Set();
     txns.add(value);
     this.transactionsBySecurityId_.set(value.security_id, txns);
   }
 
+  public hasPendingTransactions(): boolean {
+    return this.pendingTransactionsBySecurityId_.length > 0;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private checkPendingTransactions(value: any): void {
+    const txn = this.pendingTransactionsBySecurityId_.findIndex(
+      (item) =>
+        item.object_type === "TX_PLAN_SECURITY_RETRACTION" &&
+        item.id === value.id
+    );
+    if (txn !== -1) {
+      this.pendingTransactionsBySecurityId_.splice(txn, 1);
+    }
+  }
+
```

Comment:
**Question:** Are we still using these or can we delete?

Reply:
Will delete it in the next PR